### PR TITLE
Add MonadReader/MonadState/MonadWriter instances

### PR DESCRIPTION
**Pull request by 1Jajen1**
Adds instances for those typeclasses for every mtl type that supports it.

### CODE REVIEW

File: modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/WriterTTest.kt [hunk 1]
```diff
@@ -108,12 +114,18 @@ class WriterTTest : UnitSpec() {
...
+        WriterT.monadState<StateTPartialOf<ForId, Int>, String, Int>(String.monoid(), StateT.monadState(Id.monad())),
+        WriterT.genK(StateT.genK(Id.genK(), Gen.int()), Gen.string()),
+        WriterT.eqK(StateT.eqK(Id.eqK(), Int.eq(), Id.monad(), 0), String.eq())
       )
     )
   }
 }
 
-private fun <F, W> WriterT.Companion.genK(
+internal fun <F, W> WriterT.Companion.genK(
```
Comment (nomisRev) [inline modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/WriterTTest.kt]: These should publicly live in an `arrow-test`(-mtl) module?

Comment (1Jajen1) [reply modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/WriterTTest.kt]: I think we could get away with putting them in `arrow-test` for now. That project depends on mtl-data, recursion-data, fx etc already anyway.

I think at some point that should be split apart, but that could be quite a big change as lots of tests could transiently depend on that behaviour...

Comment (1Jajen1) [reply modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/WriterTTest.kt]: Moved them to `arrow-test` for now

**Comment by 1Jajen1** [comment]
Re-running tests because the fx tests continue to be flaky :/

**Comment by 1Jajen1** [comment]
This did some weird stuff during the last merge... Somehow committed a bunch of files that are not supposed to be there, will clean that up tomorrow. Please don't merge^^

**Comment by 1Jajen1** [comment]
Fixed the problems, this should be good to go now.

**Comment by rachelcarmena** [comment]
Hi! Libraries have been moved to the new repositories. Please, find the steps here: https://github.com/arrow-kt/arrow/blob/master/docs/move-to-multi-repo/how-to-move-a-pr.md and let me know if I can help you to move this PR to the correspondent repository. Thanks for your collaboration!!

**Pull request closed without merging**